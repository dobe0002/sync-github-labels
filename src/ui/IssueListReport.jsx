/* eslint-disable react/jsx-one-expression-per-line */
import React from 'react';
import PropTypes from 'prop-types';

const IssueListReport = props => {
  /* *********************************************** */
  const labelInfo = label => {
    return (
      <div>
        <h2>Label: {label.name}</h2>
        <div>
          <p>Total uses: {label.total}</p>
          <p>
            Found in {label.repos} out of {label.reposchecked} repos checked
          </p>
        </div>
      </div>
    );
  };
  /* *********************************************** */
  const issueInfo = issue => {
    return (
      <li key={issue.url} className='issue'>
        <a href={issue.url}>{issue.name}</a> (#
        {issue.number}) {issue.open !== true && 'CLOSED'}
      </li>
    );
  };

  /* *********************************************** */
  const repoInfo = repo => {
    const key = `${repo.org}${repo.name}`;
    return (
      <div key={key} className='repo'>
        <h3>
          {repo.org}/{repo.name}
        </h3>
        <ul>{repo.issues.map(issue => issueInfo(issue))}</ul>
      </div>
    );
  };

  /* *********************************************** */
  const { reportjson } = props;
  return (
    <html lang='en'>
      <head>
        <link
          rel='stylesheet'
          href='https://stackpath.bootstrapcdn.com/bootstrap/4.4.1/css/bootstrap.min.css'
          integrity='sha384-Vkoo8x4CGsO3+Hhxv8T/Q5PaXtkKtu6ug5TOeNV6gBiFeWPGFN9MuhOf23Q9Ifjh'
          crossOrigin='anonymous'
        />
      </head>
      <body>
        <main className='container-fluid'>
          {labelInfo(reportjson.label)}
          {reportjson.repos.map(repo => repoInfo(repo))}
        </main>
      </body>
    </html>
  );
};

IssueListReport.propTypes = {
  reportjson: PropTypes.shape({
    label: PropTypes.object,
    repos: PropTypes.array
  })
};
IssueListReport.defaultProps = {
  reportjson: { label: { name: '', total: 0, repos: 0, reposchecked: 0 } }
};
export default IssueListReport;
