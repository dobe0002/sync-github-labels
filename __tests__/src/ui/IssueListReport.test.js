/* eslint-disable react/jsx-filename-extension */
import React from 'react';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import IssueListReport from '../../../src/ui/IssueListReport';
import sampleProps from '../../../__fixtures__/issueListProps';

configure({ adapter: new Adapter() });

const { axe, toHaveNoViolations } = require('jest-axe');

expect.extend(toHaveNoViolations);

describe('Issue List Report test', () => {
  test('Accessibility check', async () => {
    const wrapper = shallow(<IssueListReport reportjson={sampleProps} />);
    const results = await axe(wrapper.html());
    // console.log('Axe violations', results.violations);
    expect(results).toHaveNoViolations();
    // console.log(wrapper.html());
  });
  test('Snapshot test', () => {
    const wrapper = shallow(<IssueListReport reportjson={sampleProps} />);
    expect(wrapper.find('.repo')).toHaveLength(3);
    expect(wrapper.find('.issue')).toHaveLength(3);
    expect(wrapper).toMatchSnapshot();
  });
});
