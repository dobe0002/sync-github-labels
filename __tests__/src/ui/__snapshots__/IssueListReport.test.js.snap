// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Issue List Report test Snapshot test 1`] = `ShallowWrapper {}`;
